fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostModelV1.java

```diff
@@ -65,7 +65,16 @@ class CostModelV1 extends PlanVisitor<Cost, PlanContext> {
     // the penalty factor is no more than BROADCAST_JOIN_SKEW_PENALTY_LIMIT
     static final double BROADCAST_JOIN_SKEW_RATIO = 30.0;
     static final double BROADCAST_JOIN_SKEW_PENALTY_LIMIT = 2.0;
-    private int beNumber = Math.max(1, ConnectContext.get().getEnv().getClusterInfo().getBackendsNumber(true));
+    private int beNumber = 1;
+
+    public CostModelV1() {
+        if (ConnectContext.get().getSessionVariable().isPlayNereidsDump()) {
+            beNumber = ConnectContext.get().getSessionVariable().getBeNumber();
+        } else {
+            beNumber = Math.max(1, ConnectContext.get().getEnv().getClusterInfo().getBackendsNumber(true));
+            ConnectContext.get().getSessionVariable().setBeNumber(beNumber);
```

Comment:
should not change session variable anytime

Reply:
get